packages/core/src/main.ts

```diff
@@ -147,14 +150,22 @@ export async function createLibp2pInstance(
   libp2p._dht._lan._network._protocol = HOPR_DHT_LAN_PROTOCOL
   libp2p._dht._lan._topologyListener._protocol = HOPR_DHT_LAN_PROTOCOL
 
-  const onConnection = libp2p.upgrader.onConnection
-
-  // @TODO implement whitelisting support
-  libp2p.upgrader.onConnection = (conn: Connection) => {
-    // if (isWhitelisted()) {
-    onConnection(conn)
-    // }
+  const onConnectionOriginal = libp2p.upgrader.onConnection
+  // check if connection is allowed
+  libp2p.upgrader.onConnection = async (conn: Connection) => {
+    const allowed = await reviewConnection(conn.remotePeer, 'libp2p peer connect')
```

Comment:
Can `reviewConnection` throw an exception? Seems like it can. Should be handled.

Reply:
Fixed